Fix ‘is_trivially_copy_constructible’ is not a member of ‘std’ error

From run_tests.sh...

```
$ test/run_tests.sh $(ls -d -1 out/webrtc*/) Release
In file included from /home/travis/build/vsimon/webrtcbuilds/out/webrtcbuilds-22893-cbaa254-linux-x64/include/api/optional.h:25:0,
                 from /home/travis/build/vsimon/webrtcbuilds/out/webrtcbuilds-22893-cbaa254-linux-x64/include/rtc_base/socket.h:28,
                 from /home/travis/build/vsimon/webrtcbuilds/out/webrtcbuilds-22893-cbaa254-linux-x64/include/rtc_base/asyncsocket.h:15,
                 from /home/travis/build/vsimon/webrtcbuilds/out/webrtcbuilds-22893-cbaa254-linux-x64/include/rtc_base/socketfactory.h:14,
                 from /home/travis/build/vsimon/webrtcbuilds/out/webrtcbuilds-22893-cbaa254-linux-x64/include/rtc_base/socketserver.h:15,
                 from /home/travis/build/vsimon/webrtcbuilds/out/webrtcbuilds-22893-cbaa254-linux-x64/include/rtc_base/messagequeue.h:30,
                 from /home/travis/build/vsimon/webrtcbuilds/out/webrtcbuilds-22893-cbaa254-linux-x64/include/rtc_base/thread.h:25,
                 from /home/travis/build/vsimon/webrtcbuilds/test/simple_app.cc:4:
/home/travis/build/vsimon/webrtcbuilds/out/webrtcbuilds-22893-cbaa254-linux-x64/include/rtc_base/sanitizer.h: In function ‘constexpr bool rtc::sanitizer_impl::IsTriviallyCopyable()’:
/home/travis/build/vsimon/webrtcbuilds/out/webrtcbuilds-22893-cbaa254-linux-x64/include/rtc_base/sanitizer.h:101:28: error: ‘is_trivially_copy_constructible’ is not a member of ‘std’
   return static_cast<bool>(std::is_trivially_copy_constructible<T>::value &&
                            ^
/home/travis/build/vsimon/webrtcbuilds/out/webrtcbuilds-22893-cbaa254-linux-x64/include/rtc_base/sanitizer.h:101:66: error: expected primary-expression before ‘>’ token
   return static_cast<bool>(std::is_trivially_copy_constructible<T>::value &&
                                                                  ^
/home/travis/build/vsimon/webrtcbuilds/out/webrtcbuilds-22893-cbaa254-linux-x64/include/rtc_base/sanitizer.h:101:67: error: ‘::value’ has not been declared
   return static_cast<bool>(std::is_trivially_copy_constructible<T>::value &&
                                                                   ^
/home/travis/build/vsimon/webrtcbuilds/out/webrtcbuilds-22893-cbaa254-linux-x64/include/rtc_base/sanitizer.h:102:29: error: ‘is_trivially_copy_assignable’ is not a member of ‘std’
                            (std::is_trivially_copy_assignable<T>::value ||
                             ^
/home/travis/build/vsimon/webrtcbuilds/out/webrtcbuilds-22893-cbaa254-linux-x64/include/rtc_base/sanitizer.h:102:64: error: expected primary-expression before ‘>’ token
                            (std::is_trivially_copy_assignable<T>::value ||
                                                                ^
/home/travis/build/vsimon/webrtcbuilds/out/webrtcbuilds-22893-cbaa254-linux-x64/include/rtc_base/sanitizer.h:102:65: error: ‘::value’ has not been declared
                            (std::is_trivially_copy_assignable<T>::value ||
                                                                 ^
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,15 +3,23 @@ sudo: required
 os:
 - linux
 - osx
+addons:
+  apt:
+    sources:
+      - ubuntu-toolchain-r-test
+    packages:
+      # Update the default 'trusty' linker and gcc version for tests to link and pass.
+      - g++-5
+      - binutils-2.26
 env:
   global:
   - LINUX_PICKLE_URL=https://www.dropbox.com/s/yclmzq1n63cminb/linux-pickle.tar.gz?dl=0
   - OSX_PICKLE_URL=https://www.dropbox.com/s/m23s3l9g1s9yw87/osx-pickle.tar.gz?dl=0
 before_script:
-# Update the default 'trusty' linker version for tests to link and pass.
+# Update the default 'trusty' linker and gcc version for tests to link and pass.
 - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
-  sudo apt-get install -y binutils-2.26 && sudo update-alternatives --install
-  /usr/bin/ld ld /usr/lib/binutils-2.26/bin/ld 90; fi
+  sudo update-alternatives --install /usr/bin/ld ld /usr/lib/binutils-2.26/bin/ld 90;
+  fi
 # To help prevent timeouts, download a 'pickled' .tar.gz archive of the linux
 # webrtc checkout. LINUX_PICKLE_URL was created with:
 # `GZIP=-9 tar zcvf linux-pickle.tar.gz .`
@@ -21,6 +29,10 @@ before_script:
 - tar xzf pickle.tar.gz -C out && rm pickle.tar.gz
 script:
 - ./build.sh -n Release -d
+# Run again since build.sh installation of dependencies sets back defaults
+- if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
+  sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-5 90 --slave /usr/bin/g++ g++ /usr/bin/g++-5;
+  fi
 - test/run_tests.sh $(ls -d -1 out/webrtc*/) Release
 # Allow failures on osx until the travis osx pipeline is more robust.
 matrix:

